src/UserContainer.java

```diff
@@ -1,19 +1,27 @@
-import java.util.HashMap;
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.List;
 
 public class UserContainer {
-   private final static Map<String, String> users = new HashMap<String, String>();
 
-    public static Map<String, String> getUsers() {
+    private static final List<User> users = new ArrayList();
+
+    public static final List<User> getUsers() {
         return users;
     }
 
-    public static boolean isUserNameAndPasswordExists(String userName, String password) {
-        boolean userExists = false;
-
-        if (users.containsKey(userName) && users.containsValue(password)) {
-            userExists = true;
+    public static boolean isUserNameAndPasswordExists(String userName, String password) throws UserNotFoundException {
+        boolean existUser = false;
+        for (int i = 0; i < users.size(); i++) {
+            User foundUser = users.get(i);
+            if (foundUser.getUsername().equals(userName)) {
+                if (foundUser.getPassword().equals(password)) {
+                    existUser = true;
```

Comment:
Do you need to go further through your users list in case you've found user?
you will iterate over ALL the list